glommio/src/sys/mod.rs

```diff
@@ -84,6 +84,13 @@ pub(crate) fn write_eventfd(eventfd: RawFd) {
     assert_eq!(ret, 8);
 }
 
+pub(crate) fn read_eventfd(eventfd: RawFd) -> io::Result<u64> {
+    let mut buf = [0u8; 8];
+    let ret = syscall!(read(eventfd, buf.as_mut_ptr() as *mut libc::c_void, 8))?;
```

Comment:
In `Reactor::new()`, we call `create_eventfd` with `OFlag::O_NONBLOCK`. That means `read()` here may return with `-EAGAIN` right? Then, the following assert may be inappropriate.